build_tools/docker/cmake_nvidia/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# An image for building IREE using CMake and testing IREE with NVIDIA GPUs.
+
+# Build using:
+# docker build --tag gcr.io/iree-oss/cmake-nvidia \
+#              build_tools/docker/cmake_nvidia/
+
+# Run interactively using the following, where IREE_WORKDIR is the path to your
+# local dev environment:
+# docker run -it --rm --entrypoint bash \
+#   --volume "${IREE_WORKDIR?}:/usr/src/iree/" \
+#   --gpus all \
+#   gcr.io/iree-oss/cmake-nvidia
+
+# Set up the image and working directory by inheriting the base
+# CMake configuration.
+# Note that we don't start from NVIDIA's docker base:
+# - nvidia/cuda (https://hub.docker.com/r/nvidia/cuda):
+#     it's.. for CUDA.
+# - nvidia/vulkan (https://hub.docker.com/r/nvidia/vulkan):
+#      does not support Ubuntu 18.04.
+# This allows to share configuration with base CMake, but it also means we need
+# to MATCH the driver version between the host machine and the docker image.
+FROM gcr.io/iree-oss/cmake
+
+# Additionally, we need to install the Vulkan SDK and the NVIDIA Vulkan driver.
+
+ARG VULKAN_SDK_VERSION=1.2.141
+
+# Disable apt-key parse waring.
+ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1
```

Comment:
Is this actually necessary? I think that's only required if you're parsing the apt-key output, which we're not here

Reply:
Yeah otherwise I see "Warning: apt-key output should not be parsed (stdout is not a terminal)" due to adding key from lunarg.

Reply:
Weird... when I run the command locally I don't see that error, but I do when building the docker container. I guess despite what the warning says, what it's actually complaining about is usage within a script? I'm worried that the warning is there for a reason:

From the apt-key docs:

> the programmatic usage (especially in package maintainer scripts!) is strongly discouraged.

It would be nice to figure out how to avoid this. Maybe it's better to to not use apt, for instance?

Anyway, let's not block on this.